client/src/components/Attachment/UploadAttachment.js

```diff
@@ -0,0 +1,175 @@
+    createAttachment(attachment: $attachment)
+  }
+`
+
+const GQL_UPDATE_ATTACHMENT = gql`
+  mutation ($attachment: AttachmentInput!) {
+    updateAttachment(attachment: $attachment)
+  }
+`
+
+const UploadAttachment = ({ getRelatedObject, edit, saveRelatedObject }) => {
+  const [error, setError] = useState(null)
+  const [remove, setRemove] = useState(false)
+  const [uploadedList, setUploadedList] = useState([])
+  const relatedObject = getRelatedObject()
+
+  const handleUploadAttachment = file => {
+    setUploadedList(current => [...current, file])
+    toast.success("Your document has been uploaded")
+  }
+
+  const attachmentSave = async e => {
+    const file = e.target.files[0]
+    const selectedAttachment = new Attachment({
+      fileName: file.name,
+      mimeType: file.type,
+      contentLength: file.size,
+      attachmentRelatedObjects: [
+        {
+          relatedObjectType: relatedObject.type,
+          relatedObjectUuid: relatedObject.uuid
+        }
+      ],
+      classification:
+        Settings.fields.attachment.classification.choices.UNDEFINED.value
+    })
+
+    return save(selectedAttachment, relatedObject.uuid, false)
+      .then(response => {
+        selectedAttachment.uuid = response.createAttachment
+        const [authHeaderName, authHeaderValue] = API._getAuthHeader()
+        const toastId = `uploadProgress.${selectedAttachment.uuid}`
+        toast.info(`Upload of ${selectedAttachment.fileName} in progress`, {
+          toastId,
+          autoClose: false,
+          closeOnClick: false,
+          pauseOnHover: false
+        })
+        return axios
+          .postForm(
+            `/api/attachment/uploadAttachmentContent/${selectedAttachment.uuid}`,
+            { file },
+            {
+              headers: { [authHeaderName]: authHeaderValue },
+              onUploadProgress: progressEvent => {
+                if (progressEvent.progress === 1) {
+                  toast.update(toastId, {
+                    render: `Processing uploaded attachment ${selectedAttachment.fileName}`
+                  })
+                } else {
+                  toast.update(toastId, {
+                    progress: progressEvent.progress
+                  })
+                }
+              }
+            }
+          )
+          .then(() => {
+            toast.done(toastId)
+            handleUploadAttachment(selectedAttachment)
+          })
+          .catch(() => {
+            toast.dismiss(toastId)
+            toast.error(
+              `Attachment content upload failed for ${selectedAttachment.fileName}`
+            )
+          })
+      })
+      .catch(error =>
+        toast.error(
+          `Attachment upload for ${selectedAttachment.fileName} failed: ${error.message}`
+        )
+      )
+  }
+
+  const handleFileEvent = async e => {
+    // Control related object has an uuid or not
+    if (!relatedObject.uuid && relatedObject.type === "reports") {
```

Comment:
First of all, there's a constant you should use instead of hard-coded `"reports"`, but second and more importantly: why?? There really should be no difference in uploading attachments whatever the related object's type is…